common/dynamicconfig/file_based_client.go

```diff
@@ -117,7 +156,94 @@ func (fc *fileBasedClient) update() error {
 		return fmt.Errorf("unable to decode dynamic config: %w", err)
 	}
 
-	return fc.storeValues(newValues)
+	oldValues := fc.basicClient.getValues()
+	err = fc.storeValues(newValues)
+	newStoredValues := fc.getValues()
+
+	fc.logDiff(oldValues, newStoredValues)
+
+	return err
+}
+
+func (fc *fileBasedClient) logDiff(old configValueMap, new configValueMap) {
```

Comment:
can you move logDiff (and related methods) to basicClient so if we ever need other client type they can be shared.